src/app/core/store/collections/collections.reducer.ts

```diff
@@ -55,28 +58,39 @@ export function collectionsReducer(state: CollectionsState = initialCollectionsS
   }
 }
 
+function onCreateAttributeSuccess(state: CollectionsState, action: CollectionsAction.CreateAttributeSuccess): CollectionsState {
+  let attributes = state.entities[action.payload.collectionId].attributes.slice();
+  attributes.push(action.payload.attribute);
```

Comment:
I am not sure but direct `concat()` instead of `slice()` and `push()` might be faster.